torch/csrc/jit/script/python_sugared_value.cpp

```diff
@@ -330,6 +330,35 @@ std::vector<std::shared_ptr<SugaredValue>> ModuleValue::asTuple(
   return result;
 }
 
+void ModuleValue::setAttr(
```

Comment:
This code is essentially a reimplementation of setAttr on the SimpleValue class when the type is a class.  But `self_` is already a class. We should change ModuleValue so that it inherits from SimpleValue. Then it would also inherit the correct definition of setAttr and no code would be needed here.